python/src/numerous/collection/numerous_collection.py

```diff
@@ -68,6 +69,78 @@ def document(self, key: str) -> NumerousDocument:
 
         return numerous_document
 
+    def file(self, key: str) -> NumerousFile:
+        """
+        Get or create a file by key.
+
+        Attributes
+        ----------
+        key (str): The key of the file.
+
+        """
+        numerous_file = self._client.get_collection_file(self.id, key)
+        if numerous_file is None:
+            msg = "Failed to retrieve or create the file."
+            raise ValueError(msg)
+
+        return numerous_file
+
+    def save_file(self, file_key: str, file_data: str) -> None:
+        """
+        Save data to a file in the collection.
+
+        If the file with the specified key already exists,
+        it will be overwritten with the new data.
+
+        Args:
+        ----
+            file_key (str): The key of the file to save or update.
+            file_data (str): The data to be written to the file.
+
+        Raises:
+        ------
+            ValueError: If the file cannot be created or saved.
+
+        """
+        numerous_file = self.file(file_key)
+        numerous_file.save(file_data)
+
+    def files(
+        self, tag_key: Optional[str] = None, tag_value: Optional[str] = None
+    ) -> Iterator[NumerousFile]:
+        """
+        Retrieve files from the collection, filtered by a tag key and value.
+
+        Parameters
+        ----------
+        tag_key : Optional[str]
+            The key of the tag used to filter files (optional).
+        tag_value : Optional[str]
+            The value of the tag used to filter files (optional).
+
+        Yields
+        ------
+        NumerousFile
+            Yields NumerousFile objects from the collection.
+
+        """
```

Comment:
We use google doc strings - please reformat the docstrings to match :)

https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings

E.g. no dashes, and indentation, and other keywords (Args instead of Parameters)

For example:

```suggestion
        """
        Retrieve files from the collection, filtered by a tag key and value.

        Args:
            tag_key: The key of the tag used to filter files (optional).
            tag_value: The value of the tag used to filter files (optional).

        Yields:
            Files from the collection.

        """
```